lib_sw_pll/src/sw_pll_common.h

```diff
@@ -17,6 +17,11 @@ typedef int32_t sw_pll_15q16_t; // Type for 15.16 signed fixed point
 #define SW_PLL_15Q16(val) ((sw_pll_15q16_t)((float)val * (1 << SW_PLL_NUM_FRAC_BITS)))
 #define SW_PLL_NUM_LUT_ENTRIES(lut_array) (sizeof(lut_array) / sizeof(lut_array[0]))
 
+// This is just here to catch an error and provide useful info if you happen to forget to include from XC properly
+typedef struct xc_check{
+    int *xc_check; // If you see this error, then you need to extern "C"{} the sw_pll include in your XC file.
+} xc_check;
```

Comment:
This looks like something that should go into a tips & tricks document somewhere, along with an explanation of how it works and when to use it.